lib/pos_blockchain.go

```diff
@@ -1412,12 +1411,21 @@ func (bc *Blockchain) shouldReorg(blockNode *BlockNode, currentView uint64) bool
 	return blockNode.Header.ProposedInView >= currentView
 }
 
-// addBlockToBestChain adds the block to the best chain.
-func (bc *Blockchain) addBlockToBestChain(blockNode *BlockNode) {
+// addBlockToBestChain adds the block as the new tip of the best chain.
+func (bc *Blockchain) addTipBlockToBestChain(blockNode *BlockNode) {
 	bc.bestChain = append(bc.bestChain, blockNode)
 	bc.bestChainMap[*blockNode.Hash] = blockNode
 }
 
+// removeTipBlockFromBestChain removes the current tip from the best chain.
+func (bc *Blockchain) removeTipBlockFromBestChain() *BlockNode {
+	// Remove the last block from the best chain.
```

Comment:
Maybe we want to add something in here that makes sure the tip isn't committed

Reply:
I'm fine leaving this method as a naive "remove tip". It's agnostic to the status of the node and is meant to be a helper that bundles the removal from the bestChain slice and bestChainMap map for convenience

Reply:
Yes if we make the other change, then i think it's okay to leave it as is and make it clear in the comments.